src/ui/TUI.java

```diff
@@ -2,48 +2,223 @@
+        System.out.println("Choose a following option by typing the number \n" +
+                "1 - Create User \n" +
+                "2 - Show Users \n" +
+                "3 - Edit User \n" +
+                "4 - Delete User \n" +
+                "5 - Exit program");
     }
 
     @Override
-    public void printMsg(String message) {
+    public void printMsg(String message)
+    {
+        System.out.println(message);
+    }
 
+    @Override
+    public void listUsers(ArrayList<User> listOfUser)
+    {
+        for (User user : listOfUser)
+        {
+            System.out.println(user.toString());
+        }
     }
 
     @Override
-    public void listUsers() {
+    public void closeProgram()
+    {
+        System.exit(0);
+    }
 
+    @Override
+    public String getUserInput()
+    {
+        return scanner.nextLine();
     }
 
     @Override
-    public void closeProgram() {
+    public User createUser()
+    {
+        String userName;
+        String initials;
+        String cpr;
+        ArrayList<String> roles;
+
+        userName = defineUsername();
+        initials = defineInitials();
+        roles = defineRoles();
+        cpr = defineCpr();
 
+        return new User(-1, userName, initials, roles, cpr, null);
     }
 
     @Override
-    public String getUserInput() {
-        return null;
+    public User editUser(User user)
+    {
+        String input;
+        String username = user.getUserName();
+        String initials = user.getInitials();
+        ArrayList<String> roles = user.getRoles();
+
+        do
+        {
+            System.out.println("Choose what you want to edit \n" +
+                    "1 - Username \n" +
+                    "2 - Initials \n" +
+                    "3 - Roles \n" +
+                    "4 - Exit");
+
+            input = scanner.nextLine();
+
+            switch(input)
+            {
+                case "1":
+                    username = defineUsername();
+                    break;
+                case "2":
+                    initials = defineInitials();
+                    break;
+                case "3":
+                    roles = defineRoles();
+                    break;
+            }
+        }
+        while (!input.equals("4"));
+
+        return new User(user.getUserId(), username, initials ,roles, user.getCpr(), user.getPassword());
     }
 
     @Override
-    public User createUser() {
-        return null;
+    public void removedUserMsg(User user)
+    {
+        System.out.println("This user was successfully removed: \n" +
+                user.getUserId() + "\n" +
+                user.getUserName() + "\n" +
+                user.getInitials() + "\n" +
+                user.getCpr());
     }
 
-    @Override
-    public User editUser(User user) {
-        return null;
+    private ArrayList<String> defineRoles()
+    {
+        ArrayList<String> roles = new ArrayList<>();
+
+        System.out.println();
+        System.out.print("User Roles \n" +
```

Comment:
Dette er også fint for nu, men vi kommer nok til at have en tabel med roller. I det tilfælde, så skal vi tage en liste som argument, som indeholder alle de tilgængelige roller i programmet.